tool/tsh/kubectl.go

```diff
@@ -101,17 +105,63 @@ func runKubectlReexec(selfExec string, args []string, collector io.Writer) error
 	return trace.Wrap(cmd.Run())
 }
 
+// wrapConfigFn wraps the rest.Config with a custom RoundTripper if the user
+// wants to sample traces.
+func wrapConfigFn(cf *CLIConf) func(c *rest.Config) *rest.Config {
+	return func(c *rest.Config) *rest.Config {
+		c.Wrap(
+			func(rt http.RoundTripper) http.RoundTripper {
+				if cf.SampleTraces {
+					// If the user wants to sample traces, wrap the transport with a trace
+					// transport.
+					return tracehttp.NewTransport(rt)
+				}
+				return rt
+			},
+		)
+		return c
+	}
+}
+
 // runKubectlCode runs the actual kubectl package code with the default options.
 // This code is only executed when `tshKubectlReexec` env is present. This happens
 // because we need to retry kubectl calls and `kubectl` calls os.Exit in multiple
 // paths.
-func runKubectlCode(args []string) {
+func runKubectlCode(cf *CLIConf, args []string) {
+	closeTracer := func() {}
+	if cf.SampleTraces {
+		provider, err := newTraceProvider(cf, "", nil)
+		if err != nil {
+			log.WithError(err).Debug("failed to set up span forwarding.")
```

Comment:
nit: Capitalize log message:

```suggestion
			log.WithError(err).Debug("Failed to set up span forwarding")
```